include/selene/exotics.h

```diff
@@ -1,32 +0,0 @@
-#pragma once
-
-#include "function.h"
-
-/*
- * Extends manipulation of primitives on the stack with more exotic
- * types
- */
-
-namespace sel {
-namespace detail {
-
-template <typename R, typename...Args>
-inline sel::function<R(Args...)> _check_get(_id<sel::function<R(Args...)>>,
-                                            lua_State *l, const int index) {
-    lua_pushvalue(l, index);
-    return sel::function<R(Args...)>{luaL_ref(l, LUA_REGISTRYINDEX), l};
-}
-
-template <typename R, typename... Args>
-inline sel::function<R(Args...)> _get(_id<sel::function<R(Args...)>> id,
-                                      lua_State *l, const int index) {
-    return _check_get(id, l, index);
-}
-
-template <typename R, typename... Args>
-inline void _push(lua_State *l, sel::function<R(Args...)> fun) {
-    fun.Push(l);
-}
-
-}
-}
```

Comment:
Removed this file because of the addition of `Reference` and `Pointer`. Since `sel::function` should be able to return the new classes adding specializations and overloads for `Reference` and `Pointer` to `exotics.h` does not work.

I found it unnecessary to keep `exotics.h`, since it actually seems to be only applicable to `sel::function`. I moved the code to `function.h`.